amazon-kinesis/1.0/modules/ROOT/pages/amazon-kinesis-connector-reference.adoc

```diff
@@ -0,0 +1,522 @@
+
+** BYTE
+** KB
+** MB
+** GB |  A data unit that qualifies the *Collection Max Size* value |  BYTE |
+| Connection Timeout a| Number |  The amount of time to wait when initially establishing the TCP connection between the connector and Amazon Kinesis server before throwing an exception if the connection fails |  30 |
+| Connection Timeout Unit a| Enumeration, one of:
+
+** NANOSECONDS
+** MICROSECONDS
+** MILLISECONDS
+** SECONDS
+** MINUTES
+** HOURS
+** DAYS |  The time unit for the *Connection Timeout* value |  SECONDS |
+| Max Connections a| Number |  Maximum number of connections to open to the backend. HTTP requests are sent in parallel over multiple connections. Setting this value too high can impact latency and consume additional resources without increasing throughput. |  24 |
+| Min Connections a| Number |  Minimum number of connections to keep open to the backend. In most cases, there is no need to increase value. |  1 |
+| Rate Limit a| Number |  Limits the number of data records per second and number of bytes per second sent to a shard. This limit can reduce bandwidth and CPU cycle wastage from sending requests that will fail from throttling. The default value, 150%, enables a single producer instance to completely saturate the allowance for a shard. To reduce throttling errors rather than completely saturate the shard, consider reducing this setting. |  150% |
+| Record TTL a| Number | Time limit for data records to be put. Records that are not successfully put within this limit fail. When setting this value, take into consideration the fact that data records still incur network latency after they leave the Amazon Kinesis Producer Library (KPL). |  30 |
+| Record TTL Unit a| Enumeration, one of:
+
+** NANOSECONDS
+** MICROSECONDS
+** MILLISECONDS
+** SECONDS
+** MINUTES
+** HOURS
+** DAYS |  The time unit for the *Record TTL* value. |  SECONDS |
+| Thread Pool Size a| Number |  Maximum number of threads with which to configure the native process thread pool. The default value, 0, enables the KPL process to choose the size of the thread pool. |  0 |
+|===
+
+==== Connection Types
+[[Config_Connection]]
+===== Kinesis Connection
+
+Connects to AWS Kinesis through the use of a user-specified access and secret key and, optionally, a role. These credentials securely sign requests to AWS services.
+
+====== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Proxy Configuration a| <<ProxyConfiguration>> |  Configuration for outbound connections through a proxy. |  |
+| TLS Configuration a| <<Tls>> |  Defines a configuration for TLS, which can be used from both the client and server sides to secure communication for the Mule app. When using the HTTPS protocol, the HTTP communication is secured using TLS or SSL. If HTTPS is configured as the protocol, then, at a minimum, the user must configure the keystore in the `tls:context` child element of the `listener-config`.  |  HTTP|
```

Comment:
Same as with Proxy, the Put Record operation does not support it. Only listener and the Checkpoint operation does.
```suggestion
| TLS Configuration a| <<Tls>> |  Protocol to use for communication. Valid values are HTTP and HTTPS. Default value is HTTP. When using HTTPS the HTTP communication is going to be secured using TLS / SSL. If HTTPS was configured as protocol then the user needs to configure at least the keystore in the tls:context child element of this listener-config. It's noteworthy that this settings is only applied to the listener (source) and the Checkpoint operation. This will have no effect on the Put Record operation.  |  HTTP|
```